### PR TITLE
Registration-Form-Basic-Functionality-Uzma

### PR DESCRIPTION
**Pull request by uzma000**
I have made basic sign in and sign up pages both for trainees and volunteers.

## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Write a summary of your change. Explain why you have made it.

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #1

## Checklist:

- [ * ] My code follows the style guidelines of this project
- [ * ] I have carefully reviewed my own code
- [ * ] I have commented my code
- [ * ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/forms/SignupForm.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+		if (!details.Name) {
+			errors.Name = "Name is required";
+		} else if (details.Name.length < 3) {
+			errors.Name = "Please enter your full name";
+		}
+		if (!details.Class) {
+			errors.Class = "Class is required";
+		} else if (details.Class.length < 3 || details.Class.length > 7) {
+			errors.Class = "Please enter the correct Class. i.e: WM3, NW4";
```
Comment (Professional-Kate) [inline client/src/pages/forms/SignupForm.js:46]: Class will be a dropdown so you won't need to validate it, every value the user can pick will be guaranteed valid

File: client/src/pages/forms/SignupForm.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+									setDetails({ ...details, Name: e.target.value })
+								}
+								value={details.Name}
+							/>
+						</div>
+						<p className="form__error">{errors.Name}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_name">Class:</label>
+							<input
+								type="text"
```
Comment (Professional-Kate) [inline client/src/pages/forms/SignupForm.js:97]: so this here would be a dropdown that's populated via fetching from the /cohorts endpoint in the backend

File: client/src/pages/forms/SignupForm.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+									setDetails({ ...details, Username: e.target.value })
+								}
+								value={details.Username}
+							/>
+						</div>
+						<p className="form__error">{errors.Username}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_name">Password:</label>
+							<input
+								type="trainee_Password"
```
Comment (Professional-Kate) [inline client/src/pages/forms/SignupForm.js:123]: I'm not sure about this type here. There is a input type of "password", I'm not sure about this type. Same as below for the confirm password input

File: client/src/pages/forms/SignupFormForVolunteers.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+						value="Sign Up"
+						className="btn"
+					/>
+				</div>
+			</form>
+		</div>
+	);
+}
+
+export default SignupFormForVolunteer;
```
Comment (Professional-Kate) [inline client/src/pages/forms/SignupFormForVolunteers.js:134]: I feel like this ``SignUpFormForVolunteers.js`` and ``SignUpForm.js`` can be the same component. You can pass in a boolen prop and based on that conditionally render the extra fields you need

**Review by Professional-Kate — CHANGES_REQUESTED** [review]
Overall really nice! I've been very pedantic and for that I appoligise, but it is overall wonderful. If you could get those small things fixed then request another review, any questions shoot them to me I'd be happy to help!

**Comment by uzma000** [comment]
Thank you Kate for reviewing. I have made the changes and sending the pull request now again. thank you

File: client/src/pages/forms/LoginForm.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+	}
+
+	useEffect(() => {
+		console.log("errors", errors);
+		if (Object.keys(errors).length === 0 && submit) {
+			setDetails(initialDetails);
+			console.log("details entered:", details);
+		}
+	},[details, errors, initialDetails, submit]);
```
[the remark this answers is not in the document]
Comment (uzma000) [reply client/src/pages/forms/LoginForm.js:28]: when i remove array, its throwing an error. I have tried through eslint but did not work as well.

File: client/src/pages/forms/SignupForm.js (hunk 1 above)
Comment (uzma000) [reply client/src/pages/forms/SignupForm.js:44]: Done :)

File: client/src/pages/forms/SignupForm.js (hunk 2 above)
Comment (uzma000) [reply client/src/pages/forms/SignupForm.js]: made dropdown menu but haven't done the fetching yet

File: client/src/pages/forms/LoginForm.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (Professional-Kate) [reply client/src/pages/forms/LoginForm.js:28]: React still confuses me so much haha, so it's fine if it works

File: client/src/pages/forms/SignupFormForVolunteers.js (hunk 4 above)
Comment (uzma000) [reply client/src/pages/forms/SignupFormForVolunteers.js:134]: will try to use props ;)

**Review by Professional-Kate — APPROVED** [review]
LGTM!

**Pull request merged**